pkg/utils/overriding/test-fixtures/patches/command-and-env/patch.yaml

```diff
@@ -4,6 +4,7 @@ commands:
     id: "commandWithTypeChanged"
   - exec:
       commandLine: "commandToReplaceCommandLineReplaced"
+      component: "mycomponent"
```

Comment:
You should not have to add the component in the `patch.yaml`, precisely because it's expected to be an override, and overrides have a dedicated schema that makes most fields optional.

But of course the `original.yaml` and `result.yaml` will need it.

Reply:
I can remove the components from the patch.yaml files before merging if you would like me to.

One side question, after installing the Red Hat Yaml checker to VSCode it shows an error if I leave the component out in these files. I added this to my settings:

    "yaml.validate": true,
    "yaml.schemas": {
        "/Users/mmulholl@us.ibm.com/devfiles/api/schemas/latest/devfile.json": ["/*.yaml"],
    }

Is there more I should add?

Reply:
If you open the VSCode workspace from this workspace definition: https://github.com/devfile/api/blob/master/devfile.api.code-workspace, then you should already have the Yaml schemas proprely configured:

```json
        "yaml.schemas": {
            "./schemas/latest/ide-targeted/devfile.json": [ "samples/devfiles/*devfile.yaml", "samples/devfile-registry/*.yaml" ],
            "./schemas/latest/ide-targeted/dev-workspace.json": [ "samples/devworkspaces/*.yaml" ],
            "./schemas/latest/ide-targeted/dev-workspace-template.json": [ "samples/devworkspace-templates/*.yaml" ],
            "./schemas/latest/ide-targeted/dev-workspace-template-spec.json": [ "pkg/utils/overriding/test-fixtures/patches/**/original.yaml", "pkg/utils/overriding/test-fixtures/patches/**/result.yaml", "pkg/utils/overriding/test-fixtures/merges/*.yaml" ],
            "./schemas/latest/ide-targeted/parent-overrides.json": [ "pkg/utils/overriding/test-fixtures/**/patch.yaml" ]
        }
```

The difference with what you have in your settings, is that the patch yaml files are  validated with the overrides schema, and not with the main one.

Reply:
Thank you. Have removed the component